### PR TITLE
Serve frontend statics through the main domain (behind CDN) on -stage…

### LINKED ISSUE #10870: Serve addons-frontend static assets through the CDN on -stage and prod
When we're ready to deploy the first batch of CDN changes to production, we need to start using the same domain to load addons-frontend static assets.

This was already done for dev in #10785, this is about stage and prod.

### PR DESCRIPTION
**Pull request by diox**
Fixes #10870

### CODE REVIEW

File: tests/unit/amo/middleware/test_cspMiddleware.js [hunk 1]
```diff
@@ -5,18 +5,7 @@ import parse from 'content-security-policy-parser';
...
-// but at the moment we only do that on dev for testing.
-// https://github.com/mozilla/addons-frontend/issues/10785
-const deployedEnvsWithStaticsOnSameDomain = ['dev'];
-
-const cdnHosts = {
-  dev: 'https://addons-amo-dev-cdn.allizom.org',
-  stage: 'https://addons-amo-cdn.allizom.org',
-  production: 'https://addons-amo.cdn.mozilla.net',
-};
+const configs = ['dev', 'stage', 'production'];
```
Comment (willdurand) [inline tests/unit/amo/middleware/test_cspMiddleware.js:8]:
```suggestion
const envs = ['dev', 'stage', 'production'];
```

?

File: tests/unit/amo/server/test_base.js [hunk 2]
```diff
@@ -165,10 +165,13 @@ describe(__filename, () => {
     });
 
     it('preconnects to addons-server CDN', async () => {
-      const config = getFakeConfig({
-        staticHost: undefined,
-        amoCDN: 'https://cdn.example.com',
-      });
+      const config = getFakeConfig(
+        {
+          staticHost: undefined,
```
Comment (willdurand) [inline tests/unit/amo/server/test_base.js:170]: Isn't that the default already?

File: tests/unit/amo/server/test_base.js [hunk 3]
```diff
@@ -165,10 +165,13 @@ describe(__filename, () => {
...
+          amoCDN: 'https://cdn.example.com',
+        },
+        { allowUnknownKeys: true },
```
Comment (willdurand) [inline tests/unit/amo/server/test_base.js:173]: Why is this needed?

File: config/default.js [hunk 4]
```diff
@@ -63,7 +62,8 @@ module.exports = {
 
   // The CDN host for AMO.
   amoCDN: addonsServerProdCDN,
-  staticHost: addonsFrontendCDN,
+  // addons-frontend statics are served by the CDN from the main domain.
+  staticHost: undefined,
```
Comment (willdurand) [inline config/default.js:66]: If this is undefined in the default config (and I don't see it defined anywhere else), could we remove the `staticHost` config parameter everywhere instead? It does not seem very useful.

File: tests/unit/amo/server/test_base.js (hunk 2 above)
Comment (diox) [reply tests/unit/amo/server/test_base.js]: ac284660baeab8985fc79a847a4e93739f26a5c3

File: config/default.js (hunk 4 above)
Comment (diox) [reply config/default.js:66]: efe8423c7737e60c3bb5ff74ed7c2888eaf0fcd4

File: tests/unit/amo/middleware/test_cspMiddleware.js (hunk 1 above)
Comment (diox) [reply tests/unit/amo/middleware/test_cspMiddleware.js]: 10b79830d45ca73bf48104c075d74def6dfce750

File: tests/unit/amo/server/test_base.js (hunk 3 above)
Comment (diox) [reply tests/unit/amo/server/test_base.js]: Because `staticHost` is `undefined` by default, so the `config.has(key)` in `getFakeConfig` fails when you try to set it :(

Comment (diox) [reply tests/unit/amo/server/test_base.js]: I removed it in a few tests in https://github.com/mozilla/addons-frontend/pull/10893/commits/e0b2c0967b9f08ed386394469ba9c118d3a32984 but it fails without it in the rest.

File: config/default.js (hunk 4 above)
Comment (willdurand) [reply config/default.js:66]: @diox why not removing ALL occurrences of `staticHost`?

Comment (diox) [reply config/default.js:66]: mmm, that's a fair point, I suppose I could go all in and completely remove the preconnect to `staticHost` too, since it should always be `undefined`. I'll investigate this.

Comment (diox) [reply config/default.js]: https://github.com/mozilla/addons-frontend/pull/10893/commits/e0200ed40c1a8f94eff82c9a98404ae68c8f3784

**Review by willdurand — COMMENTED** [review]
Looks good but could you also update `webpack.prod.config.babel.js` to get rid of the `staticHost` in this file too?

**Comment by diox** [comment]
Oops, forgot webpack. Done in https://github.com/mozilla/addons-frontend/pull/10893/commits/0918a04e41b62ea962a21c4a946863d558e45d3c

**Review by willdurand — APPROVED** [review]
